packages/runtime/src/utils/load.ts

```diff
@@ -1,180 +1,39 @@
-    url: string,
-    attrs?: Record<string, any> | undefined,
-  ) => CreateScriptHookReturn;
-}): Promise<RemoteEntryExports> {
-  const { entryExports: remoteEntryExports } = getRemoteEntryExports(
-    name,
-    globalName,
-  );
-
-  if (remoteEntryExports) {
-    return remoteEntryExports;
-  }
-
-  if (typeof document === 'undefined') {
-    return loadScriptNode(entry, {
-      attrs: { name, globalName },
-      createScriptHook,
-    })
-      .then(() => {
-        const { remoteEntryKey, entryExports } = getRemoteEntryExports(
-          name,
-          globalName,
-        );
-
-        assert(
-          entryExports,
-          `
-        Unable to use the ${name}'s '${entry}' URL with ${remoteEntryKey}'s globalName to get remoteEntry exports.
-        Possible reasons could be:\n
-        1. '${entry}' is not the correct URL, or the remoteEntry resource or name is incorrect.\n
-        2. ${remoteEntryKey} cannot be used to get remoteEntry exports in the window object.
-      `,
-        );
-
-        return entryExports;
-      })
-      .catch((e) => {
-        throw e;
-      });
-  }
-
-  return loadScript(entry, { attrs: {}, createScriptHook })
-    .then(() => {
-      const { remoteEntryKey, entryExports } = getRemoteEntryExports(
-        name,
-        globalName,
-      );
-
-      assert(
-        entryExports,
-        `
-      Unable to use the ${name}'s '${entry}' URL with ${remoteEntryKey}'s globalName to get remoteEntry exports.
-      Possible reasons could be:\n
-      1. '${entry}' is not the correct URL, or the remoteEntry resource or name is incorrect.\n
-      2. ${remoteEntryKey} cannot be used to get remoteEntry exports in the window object.
-    `,
-      );
-
-      return entryExports;
-    })
-    .catch((e) => {
-      throw e;
-    });
-}
+import { globalLoading } from '../global';
+import { Remote, RemoteEntryExports, RemoteInfo } from '../type';
+import { FederationHost } from '../core';
 
 export function getRemoteEntryUniqueKey(remoteInfo: RemoteInfo): string {
   const { entry, name } = remoteInfo;
   return composeKeyWithSeparator(name, entry);
 }
 
 export async function getRemoteEntry({
+  origin,
   remoteEntryExports,
   remoteInfo,
-  createScriptHook,
 }: {
+  origin: FederationHost;
   remoteInfo: RemoteInfo;
   remoteEntryExports?: RemoteEntryExports | undefined;
-  createScriptHook?: (
-    url: string,
-    attrs?: Record<string, any> | undefined,
-  ) => CreateScriptHookReturn;
-}): Promise<RemoteEntryExports | void> {
-  const { entry, name, type, entryGlobalName } = remoteInfo;
+}): Promise<RemoteEntryExports | false | void> {
   const uniqueKey = getRemoteEntryUniqueKey(remoteInfo);
   if (remoteEntryExports) {
     return remoteEntryExports;
   }
 
   if (!globalLoading[uniqueKey]) {
-    if (['esm', 'module'].includes(type)) {
-      globalLoading[uniqueKey] = loadEsmEntry({
-        entry,
+    const loadEntryHookRes = origin.loaderHook.lifecycle.loadEntry
```

Comment:
it will have some errors if exist multipe loadEntry hook plugin .

i recommend you can do like this : 

```
if(loadEntryHookRes){
  globalLoading[uniqueKey] = loadEntryHookRes;
  return globalLoading[uniqueKey] 
}

// ... prev internal load entry logic
```

Reply:
can you explain this one in more detail? Do we want to keep a dom fallback here instead of keeping everything nicely in the plugins? 

from my understanding: we don't want to await the `loadEntryHookRes` to check for the result, and we want to set the `globalLoading[uniqueKey]` to hook's promise immediately so we don't trigger loading the same remote entry twice.

Reply:
Yes we don't want to load the same entry twice . But image this , if you want to add `react-native-load-entry-plugin`, and both the internal `dom-plugin` and `react-native-load-entry-plugin` will be executed .

To avoid execute useless logic, we can execute user's plugin first and it has return , we will use the return directly

Reply:
I've removed the default env plugin from `runtime/core.ts` and added fallbacks for node and dom in `load.ts`. To check if there are any plugins configured I've used `hook.listeners.size` as we were always checking against `Promise` even without any plugins registered with `loadEntry` hook.